crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java

```diff
@@ -166,10 +168,8 @@ protected T internalFrom(TypeDef definition, String... ignore) {
           // if we found an annotated accessor, override the property's name if needed
           .ifPresent(a -> {
             final String fromAnnotation = (String) a.getParameters().get("value");
-            if(!updated[0].getName().equals(fromAnnotation)) {
-              updated[0] = new Property(property.getAnnotations(), property.getTypeRef(),
-                fromAnnotation, property.getComments(),
-                property.getModifiers(), property.getAttributes());
+            if(!Strings.isNullOrEmpty(fromAnnotation) && !updated[0].getName().equals(fromAnnotation)) {
```

Comment:
So the issue was that there was a JSONProperty annotation without a name? Could we add a test case for that, please?